apps/studio/components/interfaces/Storage/__tests__/CreateBucketModal.test.tsx

```diff
@@ -0,0 +1,99 @@
+      useCheckPermissions: vi.fn().mockImplementation(() => true),
+    }))
+    // useParams
+    routerMock.setCurrentUrl(`/project/default/storage/buckets`)
+    // useSelectedProject -> Project
+    addAPIMock({
+      method: `get`,
+      path: `/platform/projects/:ref`,
+      // @ts-expect-error
+      response: {
+        cloud_provider: 'localhost',
+        id: 1,
+        inserted_at: '2021-08-02T06:40:40.646Z',
+        name: 'Default Project',
+        organization_id: 1,
+        ref: 'default',
+        region: 'local',
+        status: 'ACTIVE_HEALTHY',
+      },
+    })
+    // useBucketCreateMutation
+    addAPIMock({
+      method: `post`,
+      path: `/platform/storage/:ref/buckets`,
+    })
+  })
+
+  it(`renders a dialog with a form`, async () => {
+    render(
+      <ProjectContextProvider projectRef="default">
+        <CreateBucketModal />
+      </ProjectContextProvider>
+    )
+
+    const dialogTrigger = screen.getByRole(`button`, { name: `New bucket` })
+    await userEvent.click(dialogTrigger)
+
+    await waitFor(() => {
+      expect(screen.getByRole(`dialog`)).toBeInTheDocument()
+    })
+
+    const nameInput = screen.getByLabelText(`Name of bucket`)
+    await userEvent.type(nameInput, `test`)
+
+    const standardOption = screen.getByLabelText(`Standard bucket`)
+    await userEvent.click(standardOption)
+
+    const publicToggle = screen.getByLabelText(`Public bucket`)
+    expect(publicToggle).not.toBeChecked()
+    await userEvent.click(publicToggle)
+    expect(publicToggle).toBeChecked()
+
+    const detailsTrigger = screen.getByRole(`button`, { name: `Additional configuration` })
+    expect(detailsTrigger).toHaveAttribute(`data-state`, `closed`)
+    await userEvent.click(detailsTrigger)
+    expect(detailsTrigger).toHaveAttribute(`data-state`, `open`)
+
+    const sizeLimitToggle = screen.getByLabelText(`Restrict file upload size for bucket`)
+    expect(sizeLimitToggle).not.toBeChecked()
+    await userEvent.click(sizeLimitToggle)
+    expect(sizeLimitToggle).toBeChecked()
+
+    const sizeLimitInput = screen.getByLabelText(`File size limit`)
+    expect(sizeLimitInput).toHaveValue(0)
+    await userEvent.type(sizeLimitInput, `25`)
+
+    const sizeLimitUnitSelect = screen.getByLabelText(`File size limit unit`)
+    expect(sizeLimitUnitSelect).toHaveTextContent(`bytes`)
+    await userEvent.click(sizeLimitUnitSelect)
+    const mbOption = screen.getByRole(`option`, { name: `MB` })
+    await userEvent.click(mbOption)
+    expect(sizeLimitUnitSelect).toHaveTextContent(`MB`)
+
+    const mimeTypeInput = screen.getByLabelText(`Allowed MIME types`)
+    expect(mimeTypeInput).toHaveValue(``)
+    await userEvent.type(mimeTypeInput, `image/jpeg, image/png`)
+
+    const submitButton = screen.getByRole(`button`, { name: `Create` })
+
+    fireEvent.click(submitButton)
+
+    await waitFor(() =>
+      expect(routerMock.asPath).toStrictEqual(`/project/default/storage/buckets/test`)
+    )
```

Comment:
because this component doesn't have an `onClose` callback like the other modals, we instead wait on the route navigation to validate that we've successfully created a new bucket